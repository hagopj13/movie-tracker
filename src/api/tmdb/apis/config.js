// @flow
import axios from '../axiosInstance';

export const getConfig = () => axios.get('/configuration');

export const getAllGenres = () => axios.get('/genre/movie/list');
