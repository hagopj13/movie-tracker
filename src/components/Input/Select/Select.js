// @flow
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import MuiSelect from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';

type Props = {
  options: Array<{ value: string, label: string }>,
  selectedValue: string,
  onValueChange: (value: string) => void,
};

const useStyles = makeStyles({
  root: {
    width: '100%',
  },
});

const Select = (props: Props) => {
  const { options, selectedValue, onValueChange } = props;

  const classes = useStyles();

  const handleValueChange = (event) => {
    onValueChange(event.target.value);
  };

  return (
    <MuiSelect
      className={classes.root}
      value={selectedValue}
      onChange={handleValueChange}
      variant="outlined"
      margin="dense"
    >
      {options.map((option) => (
        <MenuItem key={option.value} value={option.value}>
          {option.label}
        </MenuItem>
      ))}
    </MuiSelect>
  );
};

export default Select;
