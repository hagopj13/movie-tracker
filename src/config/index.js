import React from 'react';
import DiscoverIcon from '@material-ui/icons/Movie';
import UpcomingIcon from '@material-ui/icons/Update';
import ProfileIcon from '@material-ui/icons/Person';

export const navigationItems = [
  {
    id: 'DISCOVER',
    label: 'Discover',
    to: '/',
    icon: <DiscoverIcon />,
    requiresAuth: false,
  },
  {
    id: 'UPCOMING',
    label: 'Upcoming',
    to: '/upcoming',
    icon: <UpcomingIcon />,
    requiresAuth: false,
  },
  {
    id: 'PROFILE',
    label: 'Profile',
    to: '/profile',
    icon: <ProfileIcon />,
    requiresAuth: true,
  },
];

export const dialogTypes = {
  LOGIN: 'LOGIN',
};

export const sortOptions = [
  { value: 'popularity.desc', label: 'Popularity Descending' },
  { value: 'popularity.asc', label: 'Popularity Ascending' },
  { value: 'vote_average.desc', label: 'Rating Descending' },
  { value: 'vote_average.asc', label: 'Rating Ascending' },
  { value: 'release_date.desc', label: 'Release Date Descending' },
  { value: 'release_date.asc', label: 'Release Date Ascending' },
];
