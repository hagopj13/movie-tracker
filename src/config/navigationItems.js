import React from 'react';
import DiscoverIcon from '@material-ui/icons/Movie';
import UpcomingIcon from '@material-ui/icons/Update';
import ProfileIcon from '@material-ui/icons/Person';

export default [
  {
    id: 'DISCOVER',
    label: 'Discover',
    to: '/',
    icon: <DiscoverIcon />,
    requiresAuth: false,
  },
  {
    id: 'UPCOMING',
    label: 'Upcoming',
    to: '/upcoming',
    icon: <UpcomingIcon />,
    requiresAuth: false,
  },
  {
    id: 'PROFILE',
    label: 'Profile',
    to: '/profile',
    icon: <ProfileIcon />,
    requiresAuth: true,
  },
];
