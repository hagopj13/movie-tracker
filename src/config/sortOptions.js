export default [
  { value: 'popularity.desc', label: 'Popularity Descending' },
  { value: 'popularity.asc', label: 'Popularity Ascending' },
  { value: 'vote_average.desc', label: 'Rating Descending' },
  { value: 'vote_average.asc', label: 'Rating Ascending' },
  { value: 'release_date.desc', label: 'Release Date Descending' },
  { value: 'release_date.asc', label: 'Release Date Ascending' },
];
