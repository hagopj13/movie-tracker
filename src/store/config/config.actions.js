// @flow
import type { ImagesConfig, Genre } from 'types';

import ConfigActionTypes from './config.types';

const fetchImagesConfig = () => ({
  type: ConfigActionTypes.FETCH_IMAGES_CONFIG,
});

const fetchImagesConfigStart = () => ({
  type: ConfigActionTypes.FETCH_IMAGES_CONFIG_START,
});

const fetchImagesConfigSuccess = (imagesConfig: ImagesConfig) => ({
  type: ConfigActionTypes.FETCH_IMAGES_CONFIG_SUCCESS,
  payload: { imagesConfig },
});

const fetchImagesConfigFailure = (error: string) => ({
  type: ConfigActionTypes.FETCH_IMAGES_CONFIG_FAILURE,
  payload: { error },
});

const fetchAllGenres = () => ({
  type: ConfigActionTypes.FETCH_ALL_GENRES,
});

const fetchAllGenresStart = () => ({
  type: ConfigActionTypes.FETCH_ALL_GENRES_START,
});

const fetchAllGenresSuccess = (genres: Genre[]) => ({
  type: ConfigActionTypes.FETCH_ALL_GENRES_SUCCESS,
  payload: { genres },
});

const fetchAllGenresFailure = (error: string) => ({
  type: ConfigActionTypes.FETCH_ALL_GENRES_FAILURE,
  payload: { error },
});

export default {
  fetchImagesConfig,
  fetchImagesConfigStart,
  fetchImagesConfigSuccess,
  fetchImagesConfigFailure,
  fetchAllGenres,
  fetchAllGenresStart,
  fetchAllGenresSuccess,
  fetchAllGenresFailure,
};
