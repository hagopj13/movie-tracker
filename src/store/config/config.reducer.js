// @flow
import { handleActions } from 'redux-actions';

import type { ImagesConfig, Genre } from 'types';

import ConfigActionTypes from './config.types';

type State = {
  images: ImagesConfig,
  genres: Genre[],
};

export const initialState: State = {
  images: {
    baseImageUrl: '',
    backdropSize: '',
    posterSize: '',
    profileSize: '',
  },
  genres: [],
};

const fetchImagesConfigSuccess = (state: State, action: any) => ({
  ...state,
  images: action.payload.imagesConfig,
});

const fetchAllGenresSuccess = (state: State, action: any) => ({
  ...state,
  genres: action.payload.genres,
});

const actionHandler = {
  [ConfigActionTypes.FETCH_IMAGES_CONFIG_SUCCESS]: fetchImagesConfigSuccess,
  [ConfigActionTypes.FETCH_ALL_GENRES_SUCCESS]: fetchAllGenresSuccess,
};

export default handleActions(actionHandler, initialState);
