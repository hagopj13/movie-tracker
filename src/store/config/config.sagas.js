import { takeLatest, all, call, put } from 'redux-saga/effects';

import * as api from 'api/tmdb';
import { convertResponseToImagesConfig } from 'api/tmdb/utils';

import ConfigActionTypes from './config.types';
import configActions from './config.actions';

export function* fetchImagesConfig() {
  yield put(configActions.fetchImagesConfigStart());
  try {
    const { data } = yield call(api.getConfig);
    const imagesConfig = yield call(convertResponseToImagesConfig, data);
    yield put(configActions.fetchImagesConfigSuccess(imagesConfig));
  } catch (error) {
    yield put(configActions.fetchImagesConfigFailure(error.status_message));
  }
}

export function* fetchAllGenres() {
  yield put(configActions.fetchAllGenresStart());
  try {
    const { data } = yield call(api.getAllGenres);
    yield put(configActions.fetchAllGenresSuccess(data.genres));
  } catch (error) {
    yield put(configActions.fetchAllGenresFailure(error.status_message));
  }
}

export function* onFetchImagesConfig() {
  yield takeLatest(ConfigActionTypes.FETCH_IMAGES_CONFIG, fetchImagesConfig);
}

export function* onFetchAllGenres() {
  yield takeLatest(ConfigActionTypes.FETCH_ALL_GENRES, fetchAllGenres);
}

export default function* configSagas() {
  yield all([call(onFetchImagesConfig), call(onFetchAllGenres)]);
}
