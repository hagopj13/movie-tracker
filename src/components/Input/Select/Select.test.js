import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import MenuItem from '@material-ui/core/MenuItem';

import { sortOptions } from 'config';

import Select from './Select';

describe('Select component', () => {
  let shallow;
  let options;
  let selectedValue;
  let mockValueChange;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    options = sortOptions;
    selectedValue = options[0].value;
    mockValueChange = jest.fn();
    const mockProps = {
      options,
      selectedValue,
      onValueChange: mockValueChange,
    };
    wrapper = shallow(<Select {...mockProps} />);
  });

  it('should render the Select component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a MenuItem for each option', () => {
    expect(wrapper.find(MenuItem)).toHaveLength(options.length);
  });

  it('should render the option label as a child of MenuItem', () => {
    expect(wrapper.find(MenuItem).at(0).text()).toBe(options[0].label);
  });

  it('should pass the selectedValue prop as value prop to Select', () => {
    expect(wrapper.prop('value')).toBe(selectedValue);
  });

  it('should call the onValueChange prop when the selected option changes', () => {
    const newValue = options[1].value;
    wrapper.prop('onChange')({ target: { value: newValue } });
    expect(mockValueChange).toHaveBeenLastCalledWith(newValue);
  });
});
