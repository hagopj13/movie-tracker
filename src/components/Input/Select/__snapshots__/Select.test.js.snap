// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render the Select component correctly 1`] = `
<WithStyles(ForwardRef(Select))
  className="makeStyles-root-1"
  margin="dense"
  onChange={[Function]}
  value="popularity.desc"
  variant="outlined"
>
  <WithStyles(ForwardRef(MenuItem))
    key="popularity.desc"
    value="popularity.desc"
  >
    Popularity Descending
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="popularity.asc"
    value="popularity.asc"
  >
    Popularity Ascending
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="vote_average.desc"
    value="vote_average.desc"
  >
    Rating Descending
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="vote_average.asc"
    value="vote_average.asc"
  >
    Rating Ascending
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="release_date.desc"
    value="release_date.desc"
  >
    Release Date Descending
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="release_date.asc"
    value="release_date.asc"
  >
    Release Date Ascending
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(Select))>
`;
